runtime/common/src/main/java/org/apache/nemo/runtime/common/metric/MetricUtils.java

```diff
@@ -0,0 +1,163 @@
+   * @return the stringified execution properties.
+   */
+  static String stringifyVertexProperties(final IRDAG irdag) {
+    final StringBuilder builder = new StringBuilder();
+    irdag.getVertices().forEach(v ->
+      v.getExecutionProperties().forEachProperties(ep ->
+        epFormatter(builder, v.getId(), v.getNumericId(), ep)));
+    return builder.toString().trim();
+  }
+
+  /**
+   * Stringify edge execution properties of an IR DAG.
+   * @param irdag IR DAG to observe the edges of.
+   * @return the stringified execution properties.
+   */
+  static String stringifyEdgeProperties(final IRDAG irdag) {
+    final StringBuilder builder = new StringBuilder();
+    irdag.getVertices().forEach(v ->
+      irdag.getIncomingEdgesOf(v).forEach(e ->
+        e.getExecutionProperties().forEachProperties(ep ->
+          epFormatter(builder, e.getId(), e.getNumericId(), ep))));
+    return builder.toString().trim();
+  }
+
+  /**
+   * Formatter for execution properties.
+   * @param builder string builder to append the metrics to.
+   * @param id ID of the vertex or the edge.
+   * @param numericId numeric ID of the vertex or the edge.
+   * @param ep the execution property.
+   */
+  private static void epFormatter(final StringBuilder builder, final String id,
+                                  final Integer numericId, final ExecutionProperty<?> ep) {
+    builder.append(numericId);
+    builder.append("0");
+    final int classHashcode = ep.getClass().hashCode();
+    HASHCODE_TO_ID_AND_EP_KEY.putIfAbsent(classHashcode, Pair.of(id, ep.getClass()));
+    builder.append(classHashcode);
+
+    builder.append(":");
+    final int valueHashcode = ep.hashCode();
+    HASHCODE_TO_EXECUTION_PROPERTY.putIfAbsent(valueHashcode, ep);
+    builder.append(valueHashcode);
+    builder.append(" ");
+  }
+
+  /**
+   * Get the vertex/edge id and the execution property class (key) by the hashcode.
+   * @param hashcode the hashcode.
+   * @return the pair of id and the execution property.
+   */
+  public static Pair<String, Class<? extends ExecutionProperty>> getIdAndEPKeyByHashcode(final int hashcode) {
+    return HASHCODE_TO_ID_AND_EP_KEY.get(hashcode);
+  }
+
+  /**
+   * Get the execution property by the hashcode.
+   * @param hashcode the hashcode.
+   * @return the execution property.
+   */
+  public static ExecutionProperty<?> getExecutionPropertyByHashcode(final int hashcode) {
+    return HASHCODE_TO_EXECUTION_PROPERTY.get(hashcode);
+  }
+
+  /**
+   * Finds the project root path by looking for the travis.yml file.
+   * @return the project root path.
+   */
+  public static String fetchProjectRootPath() {
+    return recursivelyFindTravis(Paths.get(System.getProperty("user.dir")));
+  }
+
+  /**
+   * Helper method to recursively find the travis.yml file.
+   * @param path the path to search for.
+   * @return the path containing the travis.yml file.
+   */
+  private static String recursivelyFindTravis(final Path path) {
```

Comment:
I think relying on Travis looks not reasonable.
How about using `LICENSE` file or others that is more permanent..?

Reply:
That's a good idea. I'll make the change.